### PR TITLE
1653/user invite flow frontend

### PR DESCRIPTION
**Pull request by dominikx96**
# Pull Request Template

## Issue

Addresses #1653

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds the "Add user" drawer with form.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Use the newly created form to add a new user, then check if a user exists in the user table.
You can also test behavior if an email already exists (it should show a proper alert with an error message).

- [x] Desktop View
- [ ] Mobile View
- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes
- [x] I have run `yarn generate:client` if I made backend changes

### CODE REVIEW

**Comment by emilyjablonski** [comment]
Small typo in the e-mail `To complete your account creation, please the link below:` just need the word `click`.
Also it seems kinda weird to me that the email says `It will now be easier for you to start, save, and submit online applications for listings that appear on the site.` when we're inviting partners? Might be a product question bc I know that's straight from the issue.

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Small typo in the e-mail `To complete your account creation, please the link below:` just need the word `click`.
The e-mail subject should be `Welcome to the Partners Portal` and the content is a little different, but it's in the issue :)

**Comment by emilyjablonski** [comment]
When I click "Confirm my Account" from the email I'm taken to the partners sign-in page but I don't have a password yet so I'm not sure what the flow there should be. Maybe the "create a password" flow is a separate issue?

File: sites/partners/pages/users/index.tsx [hunk 1]
```diff
@@ -71,12 +80,20 @@ const Users = () => {
...
+  /* Add user drawer */
+  const [isDrawerOpen, setDrawerOpen] = useState<boolean>(false)
+
   /* Fetch user list */
   const { data: userList } = useUserList({
     page: currentPage,
     limit: itemsPerPage,
   })
 
+  /* Fetch listings */
```
Comment (emilyjablonski) [inline sites/partners/pages/users/index.tsx:92]: @seanmalbert Might eventually be worth a new query that only retrieves name, jurisdiction, and ID

File: sites/partners/src/users/FormUserAdd.tsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        return acc
+      }, [])
+      .map((id) => ({ id }))
+
+    const body = {
+      firstName,
+      lastName,
+      email,
+      roles,
+      leasingAgentInListings: leasingAgentInListings,
```
Comment (emilyjablonski) [inline sites/partners/src/users/FormUserAdd.tsx:77]: NB but don't need the colons here

**Comment by emilyjablonski** [comment]
I'm curious about my two comments above still - the content of the email is not quite right and the fact that I am redirected to sign-in without having set up a password (which is maybe a later ticket?)

**Comment by kathyccheng** [comment]
@emilyjablonski Flow for leasing agents to update their password is in a separate issue, https://github.com/bloom-housing/bloom/issues/1654

Email copy from issues re-pasted below for reference :) 

**Email Copy**

From: no-reply@housingbayarea.org
Subject Line: Welcome to the Partners Portal

Hello Name,

Welcome to the the Partners Portal on [URL]. You will now be able to manage listings and applications that you are a part of from one centralized location.

To complete your account creation, please click the link below:

Confirm my account

Thank you,
Alameda County - Housing and Community Development (HCD) Department

**Comment by dominikx96** [comment]
@emilyjablonski I updated translations and resolved conflict in the `FieldGroup` component, ready for review :)

File: backend/core/src/migration/1631110073158-add-null-jurisdiction-translation.ts [hunk 3]
```diff
@@ -7,9 +7,10 @@ export class addNullJurisdictionTranslation1631110073158 implements MigrationInt
     )
 
     translations["invite"] = {
+      hello: "Welcome to the Partners Portal",
       inviteMessage:
-        "Thank you for setting up your account on %{appUrl}. It will now be easier for you to start, save, and submit online applications for listings that appear on the site.",
-      toCompleteAccountCreation: "To complete your account creation, please the link below:",
+        "Welcome to the the Partners Portal on %{appUrl}. You will now be able to manage listings and applications that you are a part of from one centralized location.",
```
Comment (emilyjablonski) [inline backend/core/src/migration/1631110073158-add-null-jurisdiction-translation.ts:12]: One small typo here `the the`

**Comment by emilyjablonski** [comment]
I get a 400 if I try to add a user who already exists which I think is probably okay, but it means if anyone has ever created an account on the public site they we can't add them as a user on the partners site with the same email.

File: backend/core/src/migration/1631110073158-add-null-jurisdiction-translation.ts (hunk 3 above)
Comment (dominikx96) [reply backend/core/src/migration/1631110073158-add-null-jurisdiction-translation.ts]: Thanks, fixed! :)

**Review by emilyjablonski — APPROVED** [review]
LGTM apart from the existing user piece and it sounds like that's a separate PR :)

**Comment by seanmalbert** [comment]
@pbn4 , see Emily's comment above, can you fix the issue causing the 400 bad request to be thrown when a user who already exists in the system gets added. A more useful response is probably all that's needed, something like `A user with that email already exists`.

**Pull request merged**